### PR TITLE
Use `wasm-tools` workload to run benchmarks for wasm

### LINKED ISSUE #2250: WASM Microbenchmarks Not Working
Wasm Microbenchmarks are still failing after the BDN update. Two spots that pop out are a seeming failure with building, although another build of seemingly the same location succeeds after; and an inability to run a specific 'main.js' file from the browser-wasm/AppBundle folder. Respectively: 
```
// start dotnet build -c Release /p:DebugType=portable -bl:benchmarkdotnet.binlog --no-restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true /p:NuGetPackageRoot="/home/helixbot/work/A50E08AE/p/performance/artifacts/packages" in /home/helixbot/work/A50E08AE/p/performance/artifacts/bin/MicroBenchmarks/Release/net7.0/f55d017a-085a-41d6-a734-6abcdf4c2743
[2022/02/08 12:59:15][INFO] // command took 14.99s and exited with 1
```
and 
```
[2022/02/08 13:14:47][INFO] // Execute: /home/helixbot/.jsvu/v8 --expose_wasm main.js -- --run f55d017a-085a-41d6-a734-6abcdf4c2743.dll --benchmarkName Burgers.Test2 --job "PowerPlanMode=00000000-0000-0000-0000-000000000000, Runtime=Wasm, Arguments=/p:DebugType=portable,-bl:benchmarkdotnet.binlog, Toolchain=Wasm, IterationTime=250.0000 ms, MaxIterationCount=20, MinIterationCount=15, WarmupCount=1" --benchmarkId 0  in /home/helixbot/work/A50E08AE/p/performance/artifacts/bin/MicroBenchmarks/Release/net7.0/f55d017a-085a-41d6-a734-6abcdf4c2743/bin/net7.0/browser-wasm/AppBundle
[2022/02/08 13:14:47][INFO] Failed to set up high priority. Make sure you have the right permissions. Message: Permission denied
[2022/02/08 13:14:47][INFO] Error reading 'main.js'
[2022/02/08 13:14:47][INFO] No Workload Results were obtained from the run.
```

Upon further investigation, it looks like the main.js file got renamed to test-main.js. Example ls result in AppBundle:
```
dotnet.js  dotnet.timezones.blat  dotnet.wasm  icudt.dat  index.html  managed  mono-config.json  run-v8.sh  supportFiles  test-main.js
```

@naricc @SamMonoRT

### PR DESCRIPTION
**Pull request by radical**
Currently, the wasm perf pipeline uses `*LocalBuild*` targets, with a full copy of `artifacts` to build, and run the benchmarks.
And this PR changes them to use a dotnet with `wasm-tools` workload installed based on artifacts.

- this makes the process less brittle, and builds the benchmark project in the same way a user would
- removes unnecessary steps, like downloading live built libraries, and CoreCLR build
- reduces the `BrowserWasm` download needed by the perf jobs, to only contain the needed bits, instead of the whole `artifacts` dir

Note:
- this PR does not change the way blazor scenarios are run
- this installs the dotnet+workload via Wasm.Build.Tests, but this will be made more general in future PRs

Partially fixes https://github.com/dotnet/performance/issues/2250 .

### CODE REVIEW

File: eng/pipelines/coreclr/templates/perf-job.yml [hunk 1]
```diff
@@ -135,7 +135,8 @@ jobs:
...
+      - script: "mkdir $(librariesDownloadDir)/bin/wasm && unzip -o $(librariesDownloadDir)/BrowserWasm/artifacts/packages/Release/Shipping/Microsoft.NETCore.App.Runtime.Mono.browser-wasm.7.0.0-ci.nupkg data/* runtimes/* -d $(librariesDownloadDir)/bin/wasm && cp -r $(librariesDownloadDir)/BrowserWasm/artifacts/bin/dotnet-workload $(librariesDownloadDir)/bin/wasm && cp src/mono/wasm/test-main.js $(librariesDownloadDir)/bin/wasm/test-main.js && find $(librariesDownloadDir)/bin/wasm -type f -exec chmod 664 {} \\;"
+      #- script: "mkdir -p $(librariesDownloadDir)/bin/wasm/data && cp -r $(librariesDownloadDir)/BrowserWasm/dotnet-workload $(librariesDownloadDir)/bin/wasm && cp src/mono/wasm/test-main.js $(librariesDownloadDir)/bin/wasm/data/test-main.js && find $(librariesDownloadDir)/bin/wasm -type f -exec chmod 664 {} \\;"
```
Comment (DrewScoggins) [inline eng/pipelines/coreclr/templates/perf-job.yml:139]: nit: I assume we can delete this?

**Comment by DrewScoggins** [comment]
So just to make sure I am understanding this PR correctly the new flow will be:

1. Build a version of the dotnet cli that contains the wasm-tools workload as part of the installation.
2. Pass that version of the SDK to BDN using the CLI flag so we use those wasm-tools to build the benchmarks and execute them

This will be in place of what we were doing before, which was just referencing the bits directly as part of compilation in BDN. Is this right? Also adding @naricc

**Comment by radical** [comment]
> 1. Build a version of the dotnet cli that contains the wasm-tools workload as part of the installation.
> 2. Pass that version of the SDK to BDN using the CLI flag so we use those wasm-tools to build the benchmarks and execute them
> 
> This will be in place of what we were doing before, which was just referencing the bits directly as part of compilation in BDN. Is this right? Also adding @naricc

Yes. The `artifacts` folder is still getting copied in the `BrowserWasm` archive because the blazor scenarios stuff depends on it. I can clean that up, once all this is working again.

File: eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+parameters:
+  configForBuild: 'Release'
+
+steps:
+  - script: "./dotnet.sh build -p:TargetOS=Browser -p:TargetArchitecture=wasm -p:Configuration=${{ parameters.configForBuild }} /nr:false /p:TreatWarningsAsErrors=true /p:ContinuousIntegrationBuild=true src/tests/BuildWasmApps/Wasm.Build.Tests/Wasm.Build.Tests.csproj /t:InstallWorkloadUsingArtifacts"
```
Comment (naricc) [inline eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml:5]: So this is the magic part that makes a workload from the dotnet-runtime source tree and installs it?

Comment (naricc) [inline eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml:5]: I don't think I understand what is actually happening here. Is this installing a workload in some dotnet host? Where did we get the workload; from the just built dotnet-runtime?

Comment (radical) [reply eng/pipelines/coreclr/perf-wasm-prepare-artifacts-steps.yml:5]: Yeah, we install a new sdk with version from `eng/Versions.props` $(SdkVersionForWorkloadTesting)`. then install the workload manifests, and packs from the nugets generated during the runtime build, to this sdk.
Then use this sdk here. It is also used to run Wasm.Build.Tests .

**Review by lewing — APPROVED** [review]

**Comment by lewing** [comment]
@radical Is this the last change needed

**Review by naricc — APPROVED** [review]

**Comment by radical** [comment]
The current pipeline is broken because of the BDN change, till this PR is merged. If there are no objections to this then I'm going to merge it. That would put the pipeline in a better state, and we can continue making changes after that.

**Pull request merged**